File: gex/Models/Event/GameEventTeamStats.cs
using Dapper.ColumnMapper;
using gex.Code;
using System.Text.Json.Serialization;

namespace gex.Models.Event {

    [DapperColumnsMapped]
    public class GameEventTeamStats : GameEvent {

        [JsonActionLogPropertyName("teamID")]
        [ColumnMapping("team_id")]
        public int TeamID { get; set; }

        [JsonActionLogPropertyName("metalProduced")]
        [ColumnMapping("metal_produced")]
        public decimal MetalProduced { get; set; }

        [JsonActionLogPropertyName("metalUsed")]
        [ColumnMapping("metal_used")]
        public decimal MetalUsed { get; set; }

        [JsonActionLogPropertyName("metalExcess")]
        [ColumnMapping("metal_excess")]
        public decimal MetalExcess { get; set; }

        [JsonActionLogPropertyName("metalSent")]
        [ColumnMapping("metal_sent")]
        public decimal MetalSent { get; set; }

        [JsonActionLogPropertyName("metalReceived")]
        [ColumnMapping("metal_received")]
        public decimal MetalReceived { get; set; }

        [JsonActionLogPropertyName("energyProduced")]
        [ColumnMapping("energy_produced")]
        public decimal EnergyProduced { get; set; }

        [JsonActionLogPropertyName("energyReceived")]
        [ColumnMapping("energy_received")]
        public decimal EnergyReceived { get; set; }

        [JsonActionLogPropertyName("energySent")]
        [ColumnMapping("energy_sent")]
        public decimal EnergySent { get; set; }

        [JsonActionLogPropertyName("energyExcess")]
        [ColumnMapping("energy_excess")]
        public decimal EnergyExcess { get; set; }

        [JsonActionLogPropertyName("energyUsed")]
        [ColumnMapping("energy_used")]
        public decimal EnergyUsed { get; set; }

        [JsonActionLogPropertyName("damageDealt")]
        [ColumnMapping("damage_dealt")]
        public decimal DamageDealt { get; set; }

        [JsonActionLogPropertyName("damageReceived")]
        [ColumnMapping("damage_received")]
        public decimal DamageReceived { get; set; }

        [JsonActionLogPropertyName("unitsProduced")]
        [ColumnMapping("units_produced")]
        public int UnitsProduced { get; set; }

        [JsonActionLogPropertyName("unitsKilled")]
        [ColumnMapping("units_killed")]
        public int UnitsKilled { get; set; }

        [JsonActionLogPropertyName("unitsSent")]
        [ColumnMapping("units_sent")]
        public int UnitsSent { get; set; }

        [JsonActionLogPropertyName("unitsReceived")]
        [ColumnMapping("units_received")]
        public int UnitsReceived { get; set; }

        [JsonActionLogPropertyName("unitsCaptured")]
        [ColumnMapping("units_captured")]
        public decimal UnitsCaptured { get; set; }

        [JsonActionLogPropertyName("unitsOutCaptured")]
        [ColumnMapping("units_out_captured")]
        public int UnitsOutCaptured { get; set; }

    }
}

File: gex/src/model/GameEventTeamStats.ts


export class GameEventTeamsStats {

    public gameID: string = "";
    public frame: number = 0;
    public teamID: number = 0;
    public metalProduced: number = 0;
    public metalUsed: number = 0;
    public metalExcess: number = 0;
    public metalSent: number = 0;
    public metalReceived: number = 0;
    public energyProduced: number = 0;
    public energyUsed: number = 0;
    public energyExcess: number = 0;
    public energySent: number = 0;
    public energyReceived: number = 0;
    public damageDealt: number = 0;
    public damageReceived: number = 0;
    public unitsReceived: number = 0;
    public unitsKilled: number = 0;
    public unitsProduced: number = 0;
    public unitsSent: number = 0;
    public unitsCaptures: number = 0;
    public unitsOutCaptured: number = 0;

    public static parse(elem: any): GameEventTeamsStats {
        return {
            ...elem
        };
    }

}